thunder/__init__.py

```diff
@@ -448,43 +471,44 @@ def get_computation_and_inputs(*args, **kwargs):
-                    backward_traces,
-                ) = cache_entry
-                try:
-                    cs.last_prologue_execution_start = time.time_ns()
-                    if epilogue:
-                        inps, pro_to_epi = pro(*args, **kwargs)
-                    else:
-                        inps = pro(*args, **kwargs)
-                        pro_to_epi = None
-                    cs.last_prologue_execution_stop = time.time_ns()
-                except Exception as _:
-                    continue
-
-                cs.last_trace_host_tracing_start = time.time_ns()
-                cs.last_trace_host_tracing_stop = time.time_ns()
-
-                # Updates cache statistics
-                cs.cache_hits += 1
-                cs.last_traces = comp_traces
-                cs.last_interpreted_instructions = None
-                cs.last_interpreter_log = None
-                cs.last_prologue_traces = pro_traces
-                cs.last_prologue = pro
-                cs.last_prologue_transformation_start = 0
-                cs.last_prologue_transformation_stop = 0
-                cs.last_computation_transformation_start = 0
-                cs.last_computation_transformation_stop = 0
-
-                return cache_entry, inps, pro_to_epi
+                with compile_data_and_stats(cd, cs):
+                    (
+                        pro,
+                        pro_traces,
+                        comp,
+                        comp_traces,
+                        epilogue,
+                        epilogue_traces,
+                        backward_fn,
+                        backward_traces,
+                    ) = cache_entry
+                    try:
+                        cs.last_prologue_execution_start = time.time_ns()
+                        if epilogue:
+                            inps, pro_to_epi = pro(*args, **kwargs)
+                        else:
+                            inps = pro(*args, **kwargs)
+                            pro_to_epi = None
+                        cs.last_prologue_execution_stop = time.time_ns()
+                    except Exception as _:
+                        continue
+
+                    cs.last_trace_host_tracing_start = time.time_ns()
+                    cs.last_trace_host_tracing_stop = time.time_ns()
+
+                    # Updates cache statistics
+                    cs.cache_hits += 1
+                    cs.last_traces = comp_traces
+                    cs.last_interpreted_instructions = None
+                    cs.last_interpreter_log = None
+                    cs.last_prologue_traces = pro_traces
+                    cs.last_prologue = pro
+                    cs.last_prologue_transformation_start = 0
+                    cs.last_prologue_transformation_stop = 0
+                    cs.last_computation_transformation_start = 0
+                    cs.last_computation_transformation_stop = 0
+
+                    return cache_entry, inps, pro_to_epi
```

Comment:
the diff of these lines is just one indent due to the addition of `compile_data_and_stats` contextmanager?

Reply:
Yes.